### DISCUSSION TITLE
No way of nesting under pathless routes in `remix.config.js`

### DISCUSSION THREAD

**Discussion by grinkus-adapt** [remix-run/remix#7425] [Q&A] [answered]
When setting up routes in `remix.config.js`, it seems that there is no way of nesting routes under pathless routes, because there's no way of referencing a route only with a file and without a URL path. I've set up a [repro repo](https://github.com/grinkus-adapt/remix-repro_pathless-routes-in-config) to showcase the issue.

---

This is fine (`route-A` displays correctly):

![image](https://github.com/grinkus-adapt/remix-repro_pathless-routes-in-config/assets/12794717/b1069db2-c242-4ef8-8047-14f7dc300dd2)

This is also fine (alias `route-1` for `route-A` works correctly):

![image](https://github.com/grinkus-adapt/remix-repro_pathless-routes-in-config/assets/12794717/9a4de99c-333f-41e4-beb2-d4263adbfa30)

This is also fine (`route-C` displays correctly (is nested under `_invisible` route)):

![image](https://github.com/grinkus-adapt/remix-repro_pathless-routes-in-config/assets/12794717/ea75a9b5-398d-4ede-bc41-c18ff4bebfb3)

This is **wrong** (alias `route-3` for `route-C` **is not** nested under the `_invisible` route):

![image](https://github.com/grinkus-adapt/remix-repro_pathless-routes-in-config/assets/12794717/b6d0de17-ef8b-4a2b-85aa-fde3bd945cf5)

**Comment by grinkus-adapt** [comment]
React Router seems to have a different way of defining pathless routes without actually setting a `path` -- https://reactrouter.com/en/main/start/concepts#pathless-routes

```
<Routes>
  <Route path="/" element={<App />}>
    <Route index element={<Home />} />
    <Route path="teams" element={<Teams />}>
      <Route path=":teamId" element={<Team />} />
```

**Comment by brophdawg11** [comment, marked as answer]
> there's no way of referencing a route only with a file and without a URL path

The `path` parameter can `undefined` if you want a pathless route:

```js
      route(undefined, "routes/_invisible.tsx", () =>
        route(
          "/route-3",
          "routes/_invisible.route-C.tsx",
          { id: "/route-3" },
          () => {
            route("route-4", "routes/_invisible.route-C.route-D.tsx", {
              id: "/route-3/route-4",
            });
          }
        )
      );
```
